### PR TITLE
Add correct IP sorting for the ARP table

### LINKED ISSUE #947: Network overview page is not correctly sorting for IP
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

`1`

---
**Expected behaviour:**

I expect to see the IPs in a consecutive fashion:
192.168.1.1
192.168.1.2
192.168.1.3
192.168.1.4
....
192.168.1.22
192.168.1.33
192.168.1.44

**Actual behaviour:**

The IPs are sorted but the 4th octet is being sorted on it own.

192.168.1.1
192.168.1.11
192.168.1.2
192.168.1.22
192.168.1.3
192.168.1.33
192.168.1.4
192.168.1.44

image

**Steps to reproduce:**

Just access the Network Overview and sort by IP.

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

I couldn't upload the log as I'm getting this error...
 There was an error uploading your debug log.

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [X] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#git-commit---signoff))

---

**What does this PR aim to accomplish?:**

1. Fix #947 
2. Fix a bug causing the table to error if the `name` or `macVendor` fields are `NULL`.
   Note that this bug should never be triggered. We always inject (possibly empty) strings. However, manual interaction of the users with the database could trigger this as the columns are not explicitly marked `NOT NULL`.

**How does this PR accomplish the above?:**

1. Add `ip-address` `type` for `DataTables` columns
2. Check `name` and `macVendor` fields against being `null` before trying to access their `.length` property

### CODE REVIEW

File: scripts/pi-hole/js/ip-address-sorting.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+jQuery.extend( jQuery.fn.dataTableExt.oSort,
+{
+    "ip-address-pre": function ( a )
+    {
+        if (!a) { return 0 }
+        var i, item;
+        var m = a.split("."),
+            n = a.split(":"),
+            x = "",
+            xa = "";
```
Comment (Mcat12) [inline scripts/pi-hole/js/ip-address-sorting.js]: Can we avoid the very short variable names, for readability?

Comment (DL6ER) [reply scripts/pi-hole/js/ip-address-sorting.js]: I straightaway copied this from the link mentioned in the commit https://datatables.net/plug-ins/sorting/ip-address description. Do we really want to change this?

Comment (Mcat12) [reply scripts/pi-hole/js/ip-address-sorting.js]: In that case, can you add that link in a comment as being the source of the code?

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/pi-hole-4-3-2-release-notes/23852/1

**Pull request merged**